### PR TITLE
Continue patch

### PR DESCRIPTION
**Pull request by lazarusA**
continuation of https://github.com/rafaqz/Rasters.jl/pull/722

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -19,7 +19,7 @@ end
 flush_info_and_warnings()
 
 
-Logging.disable_logging(Logging.Warn)
```
Comment (asinghvi17) [inline docs/make.jl:22]: Ah I guess this is what was causing the error messages not to be shown!

**Comment by lazarusA** [comment]
@asinghvi17 could you please add the missing method? 

```
exception =
│    MethodError: expand_dimensions(::MakieCore.VertexGrid, ::Raster{Float32, 2, Tuple{X{Projected{Float64, LinRange{Float64, Int64}, ForwardOrdered, Regular{Float64}, Intervals{Start}, NoMetadata, WellKnownText{GeoFormatTypes.CRS}, Nothing, X{Colon}}}, Y{Projected{Float64, LinRange{Float64, Int64}, ReverseOrdered, Regular{Float64}, Intervals{Start}, NoMetadata, WellKnownText{GeoFormatTypes.CRS}, Nothing, Y{Colon}}}}, Tuple{Band{Categorical{Int64, UnitRange{Int64}, ForwardOrdered, NoMetadata}}}, Matrix{Float32}, Symbol, Metadata{Rasters.GDALsource, Dict{String, Any}}, Float32}) is ambiguous.
│    
│    Candidates:
│      expand_dimensions(t::Union{MakieCore.CellGrid, MakieCore.ImageLike, MakieCore.VertexGrid}, A::AbstractDimMatrix)
│        @ DimensionalDataMakie ~/.julia/packages/DimensionalData/VxDQB/ext/DimensionalDataMakie.jl:435
│      expand_dimensions(::Union{MakieCore.CellGrid, MakieCore.VertexGrid}, data::AbstractMatrix{<:Union{var"#s340", var"#s28"} where {var"#s340"<:Real, var"#s28"<:Colorant}})
│        @ Makie ~/.julia/packages/Makie/8h0bl/src/interfaces.jl:144
│    
│    Possible fix, define
│      expand_dimensions(::Union{MakieCore.CellGrid, MakieCore.VertexGrid}, ::AbstractDimMatrix{<:Union{var"#s340", var"#s28"} where {var"#s340"<:Real, var"#s28"<:Colorant}})
│    
```

File: ext/RastersMakieExt/plotrecipes.jl [hunk 2]
```diff
@@ -295,19 +295,22 @@ function Makie.convert_arguments(t::Makie.PointBased, A::AbstractRaster{<:Number
     return Makie.convert_arguments(t, _prepare_dimarray(A))
 end
 @static if isdefined(Makie, :SurfaceLike)
-
-    function Makie.convert_arguments(t::SurfaceLike, A::AbstractRaster{var"#s115", 2, D}) where {var"#s115", D<:Tuple}
+    function Makie.convert_arguments(t::SurfaceLike, A::AbstractRaster{T, 2, D}) where {T, D<:Tuple}
         return Makie.convert_arguments(t, _prepare_dimarray(A))
     end
 else # surfacelike is not a thing
+    function Makie.convert_arguments(t::Makie.VertexGrid, A::AbstractRaster{T, 2, D}) where {T, D<:Tuple}
```
Comment (lazarusA) [inline ext/RastersMakieExt/plotrecipes.jl:302]: @asinghvi17 this is not working. So, probably is wrong.

Comment (asinghvi17) [reply ext/RastersMakieExt/plotrecipes.jl:302]: what do you mean?  It works fine for me...at least in imagelike

Comment (asinghvi17) [reply ext/RastersMakieExt/plotrecipes.jl:302]: The issue is in DD

Comment (asinghvi17) [reply ext/RastersMakieExt/plotrecipes.jl:302]: It's this line

https://github.com/rafaqz/DimensionalData.jl/blob/0750b583d3ef3963c96a810e76d76a273008ac08/ext/DimensionalDataMakie.jl#L435

Comment (lazarusA) [reply ext/RastersMakieExt/plotrecipes.jl:302]: so, we need a fix in DD? mmm.

**Comment by asinghvi17** [comment]
Length of x (2161) must be equal to number of columns in z (2160)

which is at least a bit clearer

**Comment by asinghvi17** [comment]
```
help?> CellGrid()
  CellGrid() <: GridBased <: ConversionTrait

  Plots with the CellGrid trait convert their input data to (xs::Vector{Float32}, ys::Vector{Float32}, zs::Matrix{Float32}) such that (length(xs), length(ys)) == size(zs) .+ 1. After the conversion the x and y values represent the edges of
  cells corresponding to z values.

  See also: VertexGrid, ImageLike Used for: Heatmap

help?> VertexGrid
search: VertexGrid

  VertexGrid() <: GridBased <: ConversionTrait

  Plots with the VertexGrid trait convert their input data to (xs::Vector{Float32}, ys::Vector{Float32}, zs::Matrix{Float32}) such that (length(xs), length(ys)) == size(zs), or (xs::Matrix{Float32}, ys::Matrix{Float32}, zs::Matrix{Float32})
  such that size(xs) == size(ys) == size(zs).

```
and `contour` and `contourf` are VertexGrid based plots...

**Comment by asinghvi17** [comment]
It looks like the DimensionalData manual plot function override is causing this mess with `contourf`.  I'll try to clean that up.

**Comment by rafaqz** [comment]
It's inverted data, needs to be flipped by the recipe....

**Comment by asinghvi17** [comment]
Ah shit, that's not good - it means that the pure expand_dimensions / convert_arguments pipeline is failing.  That shouldn't happen since if you use any arbitrary recipe it'll call the pipeline, not the overridden method.

Looks like the error is in DImensionalData again so will check there.  Argh!

**Comment by rafaqz** [comment]
Yeah, we need Attributes setting in Makie so we can delete all of this, it's awful

**Comment by asinghvi17** [comment]
The error in contourf is that it was using `A` instead of the prepared `A1` to convert lookups.  Fixed in DD.

**Comment by rafaqz** [comment]
Omg sorry if that was me

**Comment by lazarusA** [comment]
this should work, after merging https://github.com/rafaqz/DimensionalData.jl/pull/795

**Comment by lazarusA** [comment]
@rafaqz this one is supposed to fix the docs. Now that DD is working.

**Comment by rafaqz** [comment]
Hmm whats the pipepline error? would be good to merge this

**Comment by lazarusA** [comment]
> Hmm whats the pipepline error? would be good to merge this

is mostly `doctests`, a lot!  But I don't have  all the time needed to fix them at the moment.

File: ext/RastersArchGDALExt/resample.jl [hunk 3]
```diff
@@ -50,7 +50,16 @@ function resample(A::RasterStackOrArray;
             # get crs from `to` or `x` if none was passed in
             isnothing(Rasters.crs(to)) ? Rasters.crs(A) : Rasters.crs(to)
         end
-    else
+    else # issomething(crs)
```
Comment (rafaqz) [inline ext/RastersArchGDALExt/resample.jl:53]: Can we revert this? somehow its mixed into this PR

File: ext/RastersArchGDALExt/cellsize.jl [hunk 4]
```diff
@@ -65,7 +65,7 @@ function cellsize(dims::Tuple{<:XDim, <:YDim})
             ]))
             for xb in xbnds, yb in ybnds]
     else 
-        areas = ArchGDAL.crs2transform(crs(dims), EPSG(4326)) do transform
+        areas = ArchGDAL.crs2transform(crs(dims), EPSG(4326), order = :trad) do transform
```
Comment (rafaqz) [inline ext/RastersArchGDALExt/cellsize.jl:68]: I don't think this should be here either

File: src/Rasters.jl [hunk 5]
```diff
@@ -44,6 +44,7 @@ using RecipesBase: @recipe, @series
 using Base: tail, @propagate_inbounds
 
 import GeoInterface: crs
+import Extents: Extent, extent
```
Comment (rafaqz) [inline src/Rasters.jl:47]: Looks like from another PR

File: src/Rasters.jl [hunk 6]
```diff
@@ -64,7 +65,7 @@ export missingval, boolmask, missingmask, replace_missing, replace_missing!,
        coverage, coverage!, setcrs, setmappedcrs, smapseries, cellsize
 export crs, mappedcrs, mappedindex, mappedbounds, projectedindex, projectedbounds
 export reproject, convertlookup
-
+export Extent, extent
```
Comment (rafaqz) [inline src/Rasters.jl:68]: Also from another PR

File: src/methods/mosaic.jl [hunk 7]
```diff
@@ -137,7 +137,7 @@ $EXPERIMENTAL
 """
 mosaic!(f::Function, x::RasterStackOrArray, regions::RasterStackOrArray...; kw...) = mosaic!(f, x, regions; kw...)
 function mosaic!(f::Function, A::AbstractRaster{T}, regions;
-    missingval=missingval(A), atol=_default_atol(T)
+    missingval=missingval(A), atol=maybe_eps(T)
```
Comment (rafaqz) [inline src/methods/mosaic.jl:140]: This file shouldn't change here

File: src/methods/mosaic.jl [hunk 8]
```diff
@@ -232,9 +232,3 @@ function _mosaic(span::Explicit, lookup::AbstractSampled, lookups::LookupTuple)
     newbounds = vcat(permutedims(newlower), permutedims(newupper))
     return rebuild(lookup; data=newindex, span=Explicit(newbounds))
 end
-
```
Comment (rafaqz) [inline src/methods/mosaic.jl:235]: This either

**Review by rafaqz — COMMENTED** [review]
There are a bunch of weird changes mixed into this PR...

@lazarusA can you maybe squash the whole thing locally down to one commit? then maybe merge or rebase master into that so its clean and up to date

**Comment by rafaqz** [comment]
Closed in favor of #780

**Pull request closed without merging**